### ISSUE TITLE
Error saving temp file

### ISSUE THREAD

**Issue by divan4ik** [codemix/yii2-excelexport#9]
hi!
yii2-excelexport use[ mikehaertl/php-tmpfile](https://github.com/mikehaertl/php-tmpfile) for temp files.

there is an issue about using [sys_get_temp_dir](https://github.com/mikehaertl/php-tmpfile/blob/master/src/File.php#L110) on [php.net](http://php.net/manual/ru/function.sys-get-temp-dir.php)

>This function does not account for virtualhost-specific modifications to the temp path and/or open_basedir:
```
<Virtualhost>
php_admin_value open_basedir /home/user
php_admin_value upload_tmp_dir /home/user/tmp
php_admin_value session.save_path /home/user/tmp
</Virtualhost>
```
 > Within this config it still returns /tmp

so i can't save file...

...and @mikehaertl, you have any idea?

**Comment by mikehaertl** [comment]
I've commited a fix on master. Can you help testing? You should now be able to pass constructor arguments for php-tmpfile in `fileOptions`:

```
$file = \Yii::createObject([
    'class' => 'codemix\excelexport\ExcelFile',
    'fileOptions' => [
        'directory' => '/my/tmp',
    ],
    'sheets' => [
```

You can also set `prefix` and `suffix` of the temp file if required.

**Comment by divan4ik** [comment]
[this ](https://stackoverflow.com/questions/13186069/sys-get-temp-dir-in-shared-hosting-environment) helps
```
apache2.conf  SetEnv TMPDIR /home/admin/tmp
*.php  putenv('TMPDIR='.getenv('TMPDIR'));
```
up. oh, thanks Michael, i'm on my way :running_man: 
#### up. works like a charm!

**Comment by mikehaertl** [comment]
Great. I've published 2.1.0 which includes this fix.
